modules/argocd/main.tf

```diff
@@ -91,6 +91,46 @@ resource "helm_release" "argocd_application" {
   depends_on = [module.helm_addon]
```

Comment:
```suggestion
  depends_on = [module.argocd_project]
```
You need to create the projects first, if you have applications referring to one of the projects being created below.

Reply:
Thank you for pointing this out. updated the PR.